root/static/scripts/edit/URLCleanup.js

```diff
@@ -1876,14 +1876,16 @@ const CLEANUPS = {
     match: [new RegExp('^(https?://)?www\\.muziekweb\\.(eu|nl)/', 'i')],
     type: LINK_TYPES.otherdatabases,
     clean: function (url) {
-      return url.replace(/^(?:https?:\/\/)?(?:www\.)?muziekweb\.(?:eu|nl)\/(?:[a-z]{2}\/)?Link\/([A-Z]{1,3}\d+).*$/, 'https://www.muziekweb.eu/Link/$1/');
+      return url.replace(/^(?:https?:\/\/)?(?:www\.)?muziekweb\.(?:eu|nl)\/(?:[a-z]{2}\/)?Link\/([A-Z]{1,3}\d+)(?:\/)?((CLASSICAL(?:\/COMPOSER)?|POPULAR)\/)?.*$/, 'https://www.muziekweb.nl/Link/$1/$2');
     },
     validate: function (url, id) {
       switch (id) {
         case LINK_TYPES.otherdatabases.artist:
-          return {result: /^https:\/\/www\.muziekweb\.eu\/Link\/M\d{11}\/$/.test(url)};
+          return {result: /^https:\/\/www\.muziekweb\.nl\/Link\/M\d{11}\/(CLASSICAL(?:\/COMPOSER)?|POPULAR)\/$/.test(url)};
         case LINK_TYPES.otherdatabases.release:
-          return {result: /^https:\/\/www\.muziekweb\.eu\/Link\/[A-Z]{2,3}\d{4,6}\/$/.test(url)};
+          return {result: /^https:\/\/www\.muziekweb\.nl\/Link\/[A-Z]{2,3}\d{4,6}\/$/.test(url)};
+        case LINK_TYPES.otherdatabases.work:
+          return {result: /^https:\/\/www\.muziekweb\.nl\/Link\/U\d{11}\/(CLASSICAL|POPULAR)\/$/.test(url)};
       }
```

Comment:
Common parts can be checked first, so as to check only the part specific to relationship types then:

```suggestion
      const subpath = /^https:\/\/www\.muziekweb\.nl\/Link\/(.*)\/$/.exec(url);
      if (subpath) {
      switch (id) {
        case LINK_TYPES.otherdatabases.artist:
          return {result: /^M\d{11}\/(CLASSICAL(?:\/COMPOSER)?|POPULAR)$/.test(subpath)};
        case LINK_TYPES.otherdatabases.release:
          return {result: /^[A-Z]{2,3}\d{4,6}$/.test(subpath)};
        case LINK_TYPES.otherdatabases.work:
          return {result: /^U\d{11}\/(CLASSICAL|POPULAR)$/.test(subpath)};
      }
      }
```